packages/react/src/components/button/index.tsx

```diff
@@ -0,0 +1,98 @@
+import React, {ButtonHTMLAttributes} from 'react'
+import Spinner from '../spinner'
+
+interface ButtonOptions {
```

Comment:
inspired by Chakra-UI's typed button component: https://github.com/chakra-ui/chakra-ui/blob/main/packages/button/src/button.tsx, but less clever and with decoupled styles (see #74 for reference).